### DISCUSSION TITLE
New user - Solax X3 Hybrid G4 with DTSU666 connected to inverter via modbus  (I think) - best way to integrate this setup with HA - wifi dongle vs RJ45 ethernet?

### DISCUSSION THREAD

**Discussion by darmach** [wills106/homeassistant-solax-modbus#1094] [Q&A]
Hi, as in the topic title - I'm about to become a fresh user of Solax X3 hybrid, connected with DTSU666 counter, and solax batteries.

What is the best / optimal / easiest way to integrate with HA? 

Should I use the wifi module, or order the ethernet RJ45 one - which is better to integrate, or this doesn't matter (I'd opt for a wired connection then)

**Comment by wills106** [comment]
PocketLAN doesn't provide a Modbus Connection.
PocketWiFI 3 does, but they are temperamental at times.
There is a newer PocketWiFi and LAN combined, I don't know if that provides a Modbus connection over LAN / Ethernet.
Recommended solution is a device like the Waveshare. https://homeassistant-solax-modbus.readthedocs.io/en/latest/compatible-adaptors/

**Reply by darmach** [reply]
My understanding of the setup was (I don't have the hardware in yet, and it will be installed by PV company ) that DTSU666 counter is connected via modbus to the inverter. Does it have another, secondary modbus interface I could use to connect with with HA?

**Reply by darmach** [reply]
Ok, I see that there are separate plugs for CT/Meter and COM. Each of these have their owb RS485 pins. I assume I should use com port RS485.

How about these CT/Meter connections - is CT better or DTSU666 Meter via RS485? I.e. is it an alternative or are both used?

**Reply by wills106** [reply]
> How about these CT/Meter connections - is CT better or DTSU666 Meter via RS485? I.e. is it an alternative or are both used?

Power Meter is best, so leave it as that. You use one or the other.

Yea COM is for communicating with the Inverter, so you would connect a Waveshare or similar to that.

**Reply by darmach** [reply]
@wills106 May I ask on why separate DTSU666 is better than using clamps with Solax and his "own" built in metering?

The company installing this claims it does not matter, I'd just like to make sure. I don't care about physical display and so on - what I care is to connect via modbus to "something" (I assume that something will be solar inverter or dtsu666 ) to read the data like:
 - pv produced
 - grid consumed
 - grid produced
 - battery charging/consuming
 
...to grab all data needed for HA energy dashboard for comprehensive picture.

**Reply by darmach** [reply]
I bought Waveshare USB - RS485 adapter on FT232RL chip. Here however I just found that RS485 by wave share to ethernet bridge might be "better"... https://homeassistant-solax-modbus.readthedocs.io/en/latest/modbus-adaptor-setup

Any experiences between those two? Should I return the USB and buy ethernet version?

I assume I should connect the GND to have a common ground between inverter and modbus adapter?

**Reply by darmach** [reply]
@wills106 do you have any experiences with the above? It's claimed in there that USB adapter can cause problems in HA itself.

Also there is a note on connecting only two RS485 wires, and I'm not sure if I should interpret this as "DO NOT connect GND" - as I'd expect to connect GND too to provide common ground for signal level.

**Reply by wills106** [reply]
> May I ask on why separate DTSU666 is better than using clamps with Solax and his "own" built in metering?
> 
> The company installing this claims it does not matter

I'm not an installer or an electrician. I just assumed a Power Meter would give more accurate measurements.
But it might be that they just want to save a few £££?

> Also there is a note on connecting only two RS485 wires, and I'm not sure if I should interpret this as "DO NOT connect GND" - as I'd expect to connect GND too to provide common ground for signal level.

Connecting ground on some Inverters actually introduces noise.

USB or ethernet to RS485 shouldn't make much difference. Just depends on what's easier to connect to your HA setup. They also do a POE version of the Ethernet one, so maybe even easier if you have a POE switch.
Ethernet used to be preferred as there was a lot of blocking calls in the old USB code of the Integration but I believe they have all been resolved now.

But I don't use RS485.

**Reply by darmach** [reply]
Oh, what do you use then? (For now I connected via API exposed on local network by wifi adapter with default integration, not modbus one)

**Reply by wills106** [reply]
The Gen2 X1 & the Gen3 X1/X3 Hybrids all had built in Ethernet. So can used Modbus over TCP.

They dropped the built in Ethernet on the Gen4 & 5's for whatever reason.

**Reply by steps56** [reply]
> Oh, what do you use then? (For now I connected via API exposed on local network by wifi adapter with default integration, not modbus one)

Hi,
most easy way is using the lan/wlan dongle and its ModbusTCP.
Read about supported systems/Firmware/Dongle depending at your Systems version.
But any newer since mid 2023 should be nice.
It's offering modbus, nice to integrate by wills106 brilliant integrantion.
This dongle is really not expensiv (<50 .-€).
I use it without any problems.
It's also possible to get the system working just inside your lan, cutting off this china cloud server 😎✌️.

**Reply by HA-Supi** [reply]
> > Oh, what do you use then? (For now I connected via API exposed on local network by wifi adapter with default integration, not modbus one)
> 
> Hi, most easy way is using the lan/wlan dongle and its ModbusTCP. Read about supported systems/Firmware/Dongle depending at your Systems version. But any newer since mid 2023 should be nice. It's offering modbus, nice to integrate by wills106 brilliant integrantion. This dongle is really not expensiv (<50 .-€). I use it without any problems. It's also possible to get the system working just inside your lan, cutting off this china cloud server 😎✌️.

Greetings from Stefan @ PV Forum to Stephan.. wish you a happy christmas.

As steps said use the new LAN/Wifi Adapter. But solax also Support old Generation Stick with Updates. My wifi 2.0 plus Sticl with modbus TCP is running now with FW 18.03 .. (started with 5.xx) 
Thanks Wills addon there are many Users in Germany.. 

https://www.photovoltaikforum.com/wissen/blog-entry-list/45-einstieg-in-die-qcells-q-home-ess-hyb-g3-bzw-solax-x3-g4-welt-gesammeltes-wissen/
